### ISSUE TITLE
Is it possible to invoke the service method by reading it from the soap header?

### ISSUE THREAD

**Issue by Blackleones** [DigDes/SoapCore#863]
The soap message that I am trying to consume does not have the action name inside the envelope's body but it is described in the `<action>` inside the header.

Is there a possible way on how to let soapcore determines the method to invoke reading it from the action tag inside the soap header?

```xml
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:wsa="http://www.w3.org/2005/08/addressing">
    <soap:Header>
        <wsa:Action>action_to_trigger</wsa:Action>
        <wsa:MessageID>message_id</wsa:MessageID>
    </soap:Header>
    <soap:Body>
        ...
    </soap:Body>
</soap:Envelope>    
```

**Comment by Blackleones** [comment]
Actually, I have found a workaround using another middleware in front of SoapCore but I wonder if there is a better solution 

```csharp
public class CustomPreSoapMiddleware
{
    private readonly RequestDelegate _next;
    private const string InterestedPath = "/Service.asmx";
    private const string InterestedTag = "wsa:Action";
    public CustomPreSoapMiddleware(RequestDelegate next)
    {
        _next = next;
    }

    public async Task InvokeAsync(HttpContext context)
    {
        var req = context.Request;
        
        if (req.Method.ToLower() == "post" && req.Path.Equals(InterestedPath))
        {
            req.EnableBuffering();

            using var reader = new StreamReader(req.Body, encoding: Encoding.UTF8, detectEncodingFromByteOrderMarks: false, leaveOpen: true);
            
            var body = await reader.ReadToEndAsync();
            
            XmlDocument xd = new XmlDocument();
            xd.LoadXml(body);

            TrySetSoapAction(req, xd);
            
            req.Body.Position = 0;
        }
        
        await _next(context);
    }

    private bool TrySetSoapAction(HttpRequest request, XmlDocument xd)
    {
        var action = xd.GetElementsByTagName(InterestedTag).Cast<XmlNode>().FirstOrDefault();

        if (action is not null && !string.IsNullOrEmpty(action.InnerText))
        {
            request.Headers["SOAPAction"] = action.InnerText;
            return true;
        }

        return false;
    }
}
```

**Comment by andersjonsson** [comment]
For now the workaround seems to be the only way.
Shouldn't be too hard to add though, if you feel up for adding this functionality in a PR
Take a look at
https://github.com/DigDes/SoapCore/blob/3545580821eda19dd0ae332323329e22a99ea0d5/src/SoapCore/HeadersHelper.cs#L16

That method is responsible for finding the proper SoapAction.
Right now it seems to take the name of the first tag in the message body, as a fallback.

I'd introduce a step between reading the httpheader and the current fallback, that uses requestMsg.Headers to try and find the SoapAction from a soap header. 
That should do the trick.

**Comment by andersjonsson** [comment]
Fixed by #866
